website/addons/box/routes.py

```diff
@@ -0,0 +1,190 @@
+        ),
+
+        Rule(
+            [
+                '/project/<pid>/box/oauth/',
+                '/project/<pid>/node/<nid>/box/oauth/',
+            ],
+            'get',
+            views.auth.box_oauth_start,
+            json_renderer,
+        ),
+    ],
+    'prefix': '/api/v1'
+}
+
+web_routes = {
+    'rules': [
+
+        ##### View file #####
+        Rule(
+            [
+                '/project/<pid>/box/files/<path:path>',
+                '/project/<pid>/node/<nid>/box/files/<path:path>',
+            ],
+            'get',
+            views.crud.box_view_file,
+            OsfWebRenderer('../addons/box/templates/box_view_file.mako'),
+        ),
+
+
+        ##### Download file #####
+        Rule(
+            [
+                '/project/<pid>/box/files/<path:path>/download/',
+                '/project/<pid>/node/<nid>/box/files/<path:path>/download/',
+            ],
+            'get',
+            views.crud.box_download,
+            notemplate,
+        ),
+    ],
+}
+
+api_routes = {
+    'rules': [
+
+        ##### Node settings #####
+
+        Rule(
+            ['/project/<pid>/box/config/',
+            '/project/<pid>/node/<nid>/box/config/'],
+            'get',
+            views.config.box_config_get,
+            json_renderer
+        ),
+
+        Rule(
+            ['/project/<pid>/box/config/',
+            '/project/<pid>/node/<nid>/box/config/'],
+            'put',
+            views.config.box_config_put,
+            json_renderer
+        ),
+        Rule(
+            ['/project/<pid>/box/config/',
+            '/project/<pid>/node/<nid>/box/config/'],
+            'delete',
+            views.config.box_deauthorize,
+            json_renderer
+        ),
+
+        Rule(
+            ['/project/<pid>/box/config/import-auth/',
+            '/project/<pid>/node/<nid>/box/config/import-auth/'],
+            'put',
+            views.config.box_import_user_auth,
+            json_renderer
+        ),
+
+
+        Rule(
+            ['/project/<pid>/box/config/share/',
+            '/project/<pid>/node/<nid>/box/config/share/'],
+            'get',
+            views.config.box_get_share_emails,
+            json_renderer
+        ),
+
+        ##### CRUD #####
+
+        # Delete
+        Rule(
+            [
+                '/project/<pid>/box/files/<path:path>',
+                '/project/<pid>/node/<nid>/box/files/<path:path>',
+            ],
+            'delete',
+            views.crud.box_delete_file,
+            json_renderer
+        ),
+
+        # Upload
+        Rule(
+            [
+                '/project/<pid>/box/files/',
+                '/project/<pid>/box/files/<path:path>',
+                '/project/<pid>/node/<nid>/box/files/',
+                '/project/<pid>/node/<nid>/box/files/<path:path>',
+            ],
+            'post',
+            views.crud.box_upload,
+            json_renderer
+        ),
+
+        ##### File rendering #####
```

Comment:
File rendering and file revisions are taken care of by 0.25 and waterbutler, respectively.